lib/internal/url.js

```diff
@@ -713,6 +713,16 @@ class URL {
   toJSON() {
     return this.#context.href;
   }
+
+  static canParse(url, base = undefined) {
+    url = `${url}`;
+
+    if (base !== undefined) {
+      base = `${base}`;
+    }
+
+    return parse(url, base, () => {});
```

Comment:
It works like this. But we can make it faster. Can you add an additional function to C++ that just returns boolean? No need to pass extra strings to JS.